DuckDuckGo/Subscription/ViewModel/SubscriptionFlowViewModel.swift

```diff
@@ -131,15 +136,49 @@ final class SubscriptionFlowViewModel: ObservableObject {
                 
             }
             .store(in: &cancellables)
+        
+        subFeature.$transactionError
+            .receive(on: DispatchQueue.main)
+            .removeDuplicates()
+            .sink { [weak self] value in
+                guard let strongSelf = self else { return }
+                if let value {
+                    strongSelf.handleTransactionError(error: value)
+                }
+            }
+        .store(in: &cancellables)
        
     }
     
+    private func handleTransactionError(error: SubscriptionPagesUseSubscriptionFeature.UseSubscriptionError) {
+        switch error {
+        
+        case .purchaseFailed:
+            transactionError = .purchaseFailed
+        case .missingEntitlements:
+            transactionError = .missingEntitlements
+        case .failedToGetSubscriptionOptions:
+            transactionError = .failedToGetSubscriptionOptions
+        case .failedToSetSubscription:
+            transactionError = .failedToSetSubscription
+        case .failedToRestorePastPurchase:
+            transactionError = .failedToRestorePastPurchase
+        case .subscriptionExpired:
+            transactionError = .subscriptionExpired
+        case .hasActiveSubscription:
+            transactionError = .hasActiveSubscription
+        default:
+            transactionError = .generalError
+        }
+    }
+    
     private func setupWebViewObservers() async {
         webViewModel.$scrollPosition
             .receive(on: DispatchQueue.main)
             .sink { [weak self] value in
+                guard let strongSelf = self else { return }
                 DispatchQueue.main.async {
```

Comment:
You already use `.receive(on: DispatchQueue.main)` , is this needed?

Reply:
good catch.